pudl/init.py

```diff
@@ -546,7 +549,117 @@ def transform_ferc1(ferc1_raw_dfs,
+    # Transform EIA forms 923, 860
+    eia923_transformed_dfs = \
+        pudl.transform.eia923.transform(eia923_raw_dfs,
+                                        pudl_engine,
+                                        eia923_tables=eia923_tables,
+                                        verbose=verbose)
+    eia860_transformed_dfs = \
+        pudl.transform.eia860.transform(eia860_raw_dfs,
+                                        eia860_tables=eia860_tables,
+                                        verbose=verbose)
+    # create an eia transformed dfs dictionary
+    eia_transformed_dfs = eia860_transformed_dfs.copy()
+    eia_transformed_dfs.update(eia923_transformed_dfs.copy())
+
+    entities_dfs, eia_transformed_dfs = \
+        pudl.transform.eia.transform(eia_transformed_dfs,
+                                     eia923_years=eia923_years,
+                                     eia860_years=eia860_years,
+                                     verbose=verbose)
+    # Compile transformed dfs for loading...
+    transformed_dfs = {"Entities": entities_dfs, "EIA": eia_transformed_dfs}
+    # Load step
+    for data_source, transformed_df in transformed_dfs.items():
+        pudl.load.dict_dump_load(transformed_df,
+                                 data_source,
+                                 pudl_engine,
+                                 need_fix_inting=pc.need_fix_inting,
+                                 verbose=verbose,
+                                 csvdir=csvdir,
+                                 keep_csv=keep_csv)
+
+
+def _ETL_cems(pudl_engine, epacems_years, verbose, csvdir, keep_csv):
+    # NOTE: This a generator for raw dataframes
+    epacems_raw_dfs = pudl.extract.epacems.extract(
+        epacems_years=epacems_years, verbose=verbose)
+    # NOTE: This is a generator for transformed dataframes
+    epacems_transformed_dfs = pudl.transform.epacems.transform(
+        epacems_raw_dfs, verbose=verbose
+    )
+    if verbose:
+        print("Loading tables from EPA CEMS into PUDL:")
+    with pudl.load.BulkCopy(
+            table_name="hourly_emissions_epacems",
+            engine=pudl_engine,
+            csvdir=csvdir,
+            keep_csv=keep_csv) as loader:
+        for transformed_df_dict in epacems_transformed_dfs:
+            # There's currently only one dataframe in this dict at a time,
+            # but that could be changed if useful.
+            for yr_mo_st, transformed_df in transformed_df_dict.items():
+                # TODO: Is this dict thing useful? If not, delete
+                year, month, state = yr_mo_st
+                if verbose:
+                    print(f"    {year}-{month} for {state}...")
+                loader.add(transformed_df)
+
+    # List of indexes and constraints we need to create later, after loading
+    # See https://stackoverflow.com/a/41254430
+    # index names follow SQLAlchemy's convention ix_tablename_columnname, but
+    # this doesn't matter
+    to_create_later = [
```

Comment:
Need to be sa.DataType here.  And it looks like UniqueConstraint doesn't have a create() method, so it was breaking the tests. Maybe need to enumerate the two separately.

Reply:
Sounds good!